### PR TITLE
Refactored Inference for Parallel Processing

### PR DESCRIPTION
**Pull request by szolgyen**
The pull request has the following changes:

1. Cohort-specific code parts have been removed to keep the framework transparent. The original published version of MuTILs has been tagged and a [release](https://github.com/szolgyen/MuTILs_Panoptic/releases/tag/v1.0.0) has been added to ease looking up original cohort-specific configurations if they are needed.
    - `configs/MuTILsWSIRunConfigs.py`
    - `mutils_panoptic/MuTILsWSIRunner.py`
2. Minor optimization for internal data handling: pandas dataframe writing has been replaced with dictionary operations in the `for`-loop of `_aggregate_pixels_for_nucleus()` method of `mutils_panoptic/MuTILsInference.py`. This reduced the runtime of this part of the code to less than the half of the original version.
3. `mutils_panoptic/MuTILsInference.py` and `mutils_panoptic/MuTILsWSIRunner.py` have been refactored to adopt parallel processing of multiple ROIs of a single slide:
    - `mutils_panoptic/MuTILsWSIRunner.py`
          - Inheritance from `class MutilsInferenceRunner` has been removed for `class MuTILsWSIRunner`
          - `class MuTILsWSIRunner` orchestrates processing slides 
          - `class MuTILsWSIRunner` creates multiple CPU processes with `torch.multiprocessing`
          - Each process performs ROI-level computations on chunks of ROIs (the number of workers determines the number of chunks). For each process, a `RoiProcessor` object instance is created.
          - GPU availability check has been added with multiple options: (1) using 5 GPUs, one for each MuTILs model, (2) using a single GPU with serial processing (no parallel computation), and (3) using CPU if no GPU is available (no parallel computation).
    - `mutils_panoptic/MuTILsInference.py`
          - `class RoiProcessorConfig` dataclass has been created to store `RoiProcessor`'s configuration at process level. This enables that each process has its separate instance of ROI chunks. 
          - `class MutilsInferenceRunner` has been repurposed and renamed to `class RoiProcessor`
          - `class RoiProcessor` computes ROIs received in chunks. Each chunk of ROIs has a separate CPU process.
          - The entire preprocessing, inference, and postprocessing is done by `class RoiProcessor`'s methods.
4. Default argument setting has been removed from the constructor of `class MuTILsWSIRunner` and type hints have been added for clarity. However, the constructor still sets default values if None is given in the `MuTILsWSIRunConfigs.yaml` configuration file.
5. `mutils_panoptic/MuTILsInference.py` has been further refactored. Preprocessing codes, inference, and postprocessing codes have been separated to three child classes of `class RoiProcessor`:
     - `RoiProcessor` remains the main class of the `mutils_panoptic/MuTILsInference.py` module. The constructor of `RoiProcessor` creates instances of `RoiPreProcessor`, `RoiInferenceProcessor`, and `RoiPostProcessor`.
     - `RoiPreProcessor` takes a tile, does masking, color normalization, and prepares the batch for the model.
     - `RoiInferenceProcessor` performs the inference.
     - `RoiPostProcessor` transforms inference output to predictions, saves data, and summarize measurements.

### CODE REVIEW

**Comment by szolgyen** [comment]
1. Training codes have been reverted. 
2. A release tag has been created to keep the original, published version of MuTILs easily  accessible.

**Comment by szolgyen** [comment]
`mutils_panoptic/MuTILsInference.py` has been further refactored. Preprocessing codes, inference, and postprocessing codes have been separated to three child classes of `class RoiProcessor`:

- `RoiProcessor` remains the main class of the `mutils_panoptic/MuTILsInference.py` module. The constructor of `RoiProcessor` creates instances of `RoiPreProcessor`, `RoiInferenceProcessor`, and `RoiPostProcessor`.
- `RoiPreProcessor` takes a tile, does masking, color normalization, and prepares the batch for the model.
- `RoiInferenceProcessor` performs the inference.
- `RoiPostProcessor` transforms inference output to predictions, saves data, and summarize measurements.
